### PR TITLE
second round of docs

### PR DESCRIPTION
**Pull request by eiais**
Signed-off-by: Kyle Spiers <kyle@spiers.me>

### CODE REVIEW

**Comment by eiais** [comment]
Still need to add multi image examples

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: `signer-add`

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: `docker trust signer-add`

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: "`docker trust inspect` now lists `alice` as a valid signer:"

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: let's format as `alice`

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: "To add a new signer, `alice`, to this image: "

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: `docker trust signer-remove`

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: "To remove an existing signer, `alice`, from this image: "

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: let's format `alice`

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: "docker trust inspect now does not list `alice` as a signer:"

**Review by riyazdf — COMMENTED** [review]
thanks for tackling these!  Looking good but have some comments that were similar to the feedback we got to produce the current draft of [`docker trust inspect` docs](https://github.com/riyazdf/cli/blob/272b5a29a39582d09a22559861c1031d3d94db6d/docs/reference/commandline/trust_inspect.md)

cc @ashfall for additional review

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: could we mark up `docker trust signer-add` as code?

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: typo: remove `not`

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: the usage is `docker signer-add NAME IMAGE [IMAGE...]`

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: the usage is `docker signer-remove NAME IMAGE [IMAGE...]`

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: same comment here, could we mark up the command as code?

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: nit: extra space before alice

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [reply docs/reference/commandline/trust_signer_add.md]: Also let's change this to "`docker trust inspect` lists `alice` as a valid signer:"

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: Let's change "should now not" to "does not list"

File: docs/reference/commandline/trust_signer_add.md
Comment (eiais) [reply docs/reference/commandline/trust_signer_add.md]: And the path to the key?

File: docs/reference/commandline/trust_signer_remove.md
Comment (eiais) [reply docs/reference/commandline/trust_signer_remove.md]: There's also the --yes option

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [reply docs/reference/commandline/trust_signer_add.md]: this should match the verbatim usage from the CLI output.  So we should not mention any flags by name here, but the examples should show them in action

Comment (eiais) [reply docs/reference/commandline/trust_signer_add.md]: Usage:  docker trust signer-add [OPTIONS] NAME IMAGE [IMAGE...] 
My signer add has options

Comment (riyazdf) [reply docs/reference/commandline/trust_signer_add.md]: ah yes :+1: let's include that

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [reply docs/reference/commandline/trust_signer_remove.md]: edit: I think the usage as listed is correct.  No need to change 👍

Comment (riyazdf) [reply docs/reference/commandline/trust_signer_remove.md]: actually: `s/signer/NAME`

File: docs/reference/commandline/trust_signer_add.md
Comment (ashfall) [inline docs/reference/commandline/trust_signer_add.md]: s/signer-remove/signer-add

File: docs/reference/commandline/trust_signer_add.md
Comment (ashfall) [reply docs/reference/commandline/trust_signer_add.md]: +1 to `docker trust signer-add [OPTIONS] NAME IMAGE [IMAGE...]`

File: docs/reference/commandline/trust_signer_add.md
Comment (ashfall) [inline docs/reference/commandline/trust_signer_add.md]: s/SInitializing/ Initializing/

File: docs/reference/commandline/trust_signer_add.md
Comment (ashfall) [inline docs/reference/commandline/trust_signer_add.md]: Wouldn't this be `No signatures for example/trust-demo`?

File: docs/reference/commandline/trust_signer_add.md
Comment (ashfall) [inline docs/reference/commandline/trust_signer_add.md]: Unrelated side note: Perhaps we should reword this description when defining the subcommand as this can be used to add a signer to one or more images.

File: docs/reference/commandline/trust_signer_remove.md
Comment (ashfall) [inline docs/reference/commandline/trust_signer_remove.md]: Unrelated side note: Perhaps we should reword this description when defining the subcommand as this can be used to remove a signer from one or more images.

**Review by ashfall — COMMENTED** [review]
This is coming along great, thanks for adding more docs!

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: Change from "We can add..." to "Add `alice` as a signer with..."

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: remove "existing"

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: the docs shouldn't use "We" as per feedback in upstream PR.  Let's change to "Add `alice` as a signer to both images ..."

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: `riyaz.crt` is a little odd here.  Maybe `alice.pem`? or `alice.crt`?  We should probably make a mention about what the key argument is for in a section above, and mention that it can take a public key or certificate

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [reply docs/reference/commandline/trust_signer_remove.md]: agreed - maybe `Remove a signer from one or more images` ?   Same applies to `signer-add`

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: remove "We", change to something like "Remove `alice` as a signer with..."

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: for both `signer-add` and `signer-remove`: could we add an example of multiple-image add/remove with interspersed successes and failures?

It's important for us to highlight that these commands are best effort, and that the operation might partially succeed

Comment (eiais) [reply docs/reference/commandline/trust_signer_remove.md]: Do other docs show failures? I couldn't find an example of that.

Comment (riyazdf) [reply docs/reference/commandline/trust_signer_remove.md]: I don't think any of our `docker trust` ones do so far, but I think this case is important to highlight

File: docs/reference/commandline/trust_signer_remove.md
Comment (eiais) [inline docs/reference/commandline/trust_signer_remove.md]: The error changes depending on the error but I think showing one is enough

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [reply docs/reference/commandline/trust_signer_add.md]: I think this is still unresolved

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: note: I changed this to `keys` in the upstream PR

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: same here re: s/KeyIDs/keys

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: same here re: s/KeyIDs/keys

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: same here re: s/KeyIDs/keys

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: same here re: s/KeyIDs/keys

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: same here re: s/KeyIDs/keys

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: Let's get rid of this as a section and instead just leave the text below, as this is a natural followup from the last example

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: how about:

"`docker trust signer-add` will add signers to repositories on a best effort basis, so it will continue to add the signer to subsequent repositories if one attempt fails:"

And then cut the text starting with "Having the..." below?

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: I think this is a double, we can remove this `docker trust signer-add` command here right?

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: Let's keep an eye out for @ashfall's in-flight PR #101 since this will change - it's cool if we merge this before that merges but we'll need to update in a followup

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: Same comment here about removing this section and rewording to something like:

"`docker trust signer-remove` will remove signers to repositories on a best effort basis, so it will continue to remove the signer from subsequent repositories if one attempt fails:"  (and removing the "Having..." text block)

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: I think `eiais/test1` should be `example/unauthorized`?

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: hm, is this spacing as-is?  I'm wondering if we should (in a followup) add more newlines to make this consistent with `signer-add`?  If so, let's open a tracking issue.

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: this has changed to:
```
$ docker trust signer-remove alice example/unauthorized example/authorized

Removing signer "alice" from image example/unauthorized...
No signer alice for image example/unauthorized

Removing signer "alice" from image example/authorized...
Enter passphrase for repository key with ID c6772a0: 
Successfully removed alice from example/authorized

Error removing signer from: example/unauthorized
```

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: changed this to:

```
$ docker trust signer-add alice example/unauthorized example/authorized -k alice.crt

Adding signer "alice" to example/unauthorized...
you are not authorized to perform this operation: server returned 401.

Adding signer "alice" to example/authorized...
Enter passphrase for repository key with ID c6772a0: 
Successfully added signer: alice to example/authorized

Failed to add signer to: example/unauthorized
```

File: docs/reference/commandline/trust_signer_add.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_add.md]: last nits I promise: can we markup `docker trust signer-add` with the `code` ticks?

File: docs/reference/commandline/trust_signer_remove.md
Comment (riyazdf) [inline docs/reference/commandline/trust_signer_remove.md]: same here with `code`

**Review by riyazdf — APPROVED** [review]
LGTM after last nits, thank you for your hard work on this!

**Review by riyazdf — APPROVED** [review]

**Pull request merged**
